eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/CriterionHandler/FieldBase.php

```diff
@@ -30,11 +32,124 @@
      *
      * @param \eZ\Publish\Core\Persistence\Database\DatabaseHandler $dbHandler
      * @param \eZ\Publish\SPI\Persistence\Content\Type\Handler $contentTypeHandler
+     * @param \eZ\Publish\SPI\Persistence\Content\Language\Handler $languageHandler
      */
-    public function __construct( DatabaseHandler $dbHandler, ContentTypeHandler $contentTypeHandler )
+    public function __construct(
+        DatabaseHandler $dbHandler,
+        ContentTypeHandler $contentTypeHandler,
+        LanguageHandler $languageHandler
+    )
     {
         parent::__construct( $dbHandler );
 
         $this->contentTypeHandler = $contentTypeHandler;
+        $this->languageHandler = $languageHandler;
+    }
+
+    /**
+     * Returns the bitmask for the list languages in given $fieldFilters
+     *
+     * The method will return null if no languages are contained in $fieldFilters.
+     * Note: 'useAlwaysAvailable' filter is ignored here.
+     *
+     * @param array $fieldFilters
+     *
+     * @return null|int
+     */
+    protected function getFieldFiltersLanguageMask( array $fieldFilters )
+    {
+        if ( empty( $fieldFilters["languages"] ) )
+        {
+            return null;
+        }
+
+        $mask = 0;
+
+        foreach ( $fieldFilters["languages"] as $languageCode )
+        {
+            $mask |= $this->languageHandler->loadByLanguageCode( $languageCode )->id;
+        }
+
+        return $mask;
+    }
+
+    /**
+     * Adds filed filters condition to the WHERE clause of the given $query.
```

Comment:
Fixed in https://github.com/ezsystems/ezpublish-kernel/commit/c4d33cd4dd6f289dbc9a551c9ce96edf7e4aceb1.